orchestra/static/orchestra/common/js/orchestra.services.es6.js

```diff
@@ -97,34 +97,21 @@ export function orchestraTasks ($http) {
       return $http.get('/orchestra/api/interface/new_task_assignment/' + taskType + '/')
         .then(function (response) {
           var task = response.data
-          service.tasks.in_progress.push(task)
+          task.state = 'just_added'
+          service.tasks.push(task)
           service.tasksByAssignmentId[task.assignment_id] = task
+          return response
         })
     },
-    allTasks: function () {
-      var tasks = []
-      for (var type in this.tasks) {
-        tasks = tasks.concat(this.tasks[type])
-      }
-      return tasks
-    },
-    activeTasks: function () {
-      var tasks = []
-      for (var type in this.tasks) {
-        if (type !== 'complete') {
-          tasks = tasks.concat(this.tasks[type])
-        }
-      }
-      return tasks
-    },
+    allTasks: function () { return this.tasks },
```

Comment:
should this `function ()` be replaced with `() =>` below?

Reply:
i initially did just this to "modernize" this part of the code, but ran into an issue---the object syntax we're using to create these functions on `orchestraTasks` requires this be bound in a way that `function` definitions do, and modern `() =>` definitions do not.

i propose we modernize the whole module in a separate PR :)